### PR TITLE
Remove dead code from ocaml/xapi/sm*

### PR DESCRIPTION
**Pull request by gaborigloi**
Please let me know if I removed something that's needed.

Signed-off-by: Gabor Igloi <gabor.igloi@citrix.com>

### CODE REVIEW

File: ocaml/xapi/sm.ml [hunk 1]
```diff
@@ -98,9 +91,7 @@ let sr_detach dconf driver sr =
     (fun ()->
        debug "sr_detach" driver (sprintf "sr=%s" (Ref.string_of sr));
        let call = Sm_exec.make_call ~sr_ref:sr dconf "sr_detach" [] in
-       Sm_exec.parse_unit (Sm_exec.exec_xmlrpc (driver_filename driver) call));
-  Threadext.Mutex.execute sr_content_type_cache_m
-    (fun () -> Hashtbl.remove sr_content_type_cache sr)
+       Sm_exec.parse_unit (Sm_exec.exec_xmlrpc (driver_filename driver) call))
```
Comment (gaborigloi) [inline ocaml/xapi/sm.ml]: @mseri The only interesting change here is that I've removed a mutex that was used by dead code.

Comment (mseri) [inline ocaml/xapi/sm.ml]: Yes but you also removed the cache, so it's quite a safe change. I'd rather see if anything else (external) was using exposed functions that you removed from the interface.

Comment (gaborigloi) [inline ocaml/xapi/sm.ml]: The `xs-opam` build passed the last time I ran it: https://travis-ci.org/gaborigloi/xs-opam/builds/243744080

Comment (mseri) [inline ocaml/xapi/sm.ml]: Not all the extra packages are in `xs-opam`. Only the ones that we use or may need.

**Pull request merged**